src/plugins/auto/auto_schedule.cpp

```diff
@@ -0,0 +1,571 @@
+        }
+        devices.push_back(_loadContext[ACTUALDEVICE].deviceInfo);
+    } else {
+        // _acceleratorDevice could be the same as _cpuDevice, such as AUTO:CPU
+        if (_loadContext[ACTUALDEVICE].isAlready) {
+            devices.push_back(_loadContext[ACTUALDEVICE].deviceInfo);
+        } else {
+            // replace deviceName with workName, so schedule can select correct
+            // idleWorkerQueue
+            auto deviceInfo =  _loadContext[CPU].deviceInfo;
+            deviceInfo.deviceName = _loadContext[CPU].workName;
+            devices.push_back(std::move(deviceInfo));
+        }
+    }
+    for (auto&& device : devices) {
+        if (!preferred_device.empty() && (device.deviceName != preferred_device)) {
+            continue;
+        }
+        if (RunPipelineTask(inferPipelineTask, _idleWorkerRequests[device.deviceName], preferred_device)) {
+            return true;
+        }
+    }
+    // no vacant requests this time, storing the task to the respective queue
+    if (!preferred_device.empty()) {
+        _inferPipelineTasksDeviceSpecific[preferred_device]->push(std::move(inferPipelineTask));
+    } else {
+        _inferPipelineTasks.push(std::move(inferPipelineTask));
+    }
+    return false;
+}
+
+bool AutoSchedule::RunPipelineTask(IE::Task& inferPipelineTask,
+    NotBusyPriorityWorkerRequests& idleWorkerRequests,
+    const DeviceName& preferred_device) {
+    WorkerInferRequest* workerRequestPtr = nullptr;
+    std::pair<int, WorkerInferRequest*> worker;
+    if (idleWorkerRequests.try_pop(worker)) {
+        workerRequestPtr = worker.second;
+        IdleGuard<NotBusyPriorityWorkerRequests> idleGuard{workerRequestPtr, idleWorkerRequests};
+        _thisWorkerInferRequest = workerRequestPtr;
+        {
+            auto capturedTask = std::move(inferPipelineTask);
+            capturedTask();
+        }
+        idleGuard.Release();
+        return true;
+    }
+    return false;
+}
+
+AutoSchedule::~AutoSchedule() {
+    // this is necessary to guarantee member destroyed after getting future
+    if (_loadContext[CPU].isEnabled) {
+        _exitFlag = true;
+        _loadContext[CPU].future.wait();
+        WaitActualNetworkReady();
+        // it's necessary to wait the loading network threads to stop here.
+        _autoSContext->_plugin->executorManager()->clear("AutoDeviceAsyncLoad");
+        _executor.reset();
+    }
+    _autoSContext->_plugin->UnregisterPriority(_autoSContext->_modelPriority,
+        _loadContext[ACTUALDEVICE].deviceInfo.uniqueName);
+
+    LOG_INFO("[AUTOPLUGIN]ExecutableNetwork end");
+}
+
+
+IInferPtr AutoSchedule::CreateInferRequestImpl(
+    const std::vector<std::shared_ptr<const ov::Node>>& inputs,
+    const std::vector<std::shared_ptr<const ov::Node>>& outputs) {
+    auto num = _numRequestsCreated++;
+    IE::SoIInferRequestInternal request_to_share_blobs_with;
+    IE::RemoteContext::Ptr ctx = nullptr;
```

Comment:
double check the logic

Reply:
I think maybe we need a special pr to reduce code, not just here
maybe we can reate a function to get request_to_share_blobs_with  here

Reply:
I think maybe we need a special pr to reduce code, not just here
maybe we can reate a function to get request_to_share_blobs_with  here